src/test/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvertTest.java

```diff
@@ -92,4 +97,41 @@ void jsonToApartmentsTest() throws IOException, URISyntaxException {
     assertEquals(apartmentsRef.get(0).hashCode(), apartmentsTest.get(0).hashCode());
     assertEquals(apartmentsRef.get(1).hashCode(), apartmentsTest.get(1).hashCode());
   }
+
+  /** Test if the adapter method works */
+  @Test
+  void getAdapterTest() throws Exception {
+    JsonbConfig defaultConfig = new JsonbConfig();
+    JsonbAdapter<Apartment, JsonObject> adapter = JsonConvert.getAdapter();
+    JsonbConfig config = new JsonbConfig().withAdapters(adapter);
+    assertNotEquals(defaultConfig, config);
+    // Verification of recovered fields
+    Builder apartBuilder = new Apartment.Builder();
+    Apartment a =
+        apartBuilder
+            .setAddress("Place du Maréchal de Lattre de Tassigny, 75016 Paris")
+            .setFloorArea(1234567.89)
+            .setTitle("Paris-Dauphine - PSL")
+            .setTerrace(false)
+            .setWifi(true)
+            .setTele(true)
+            .build();
+    JsonObject json = adapter.adaptToJson(a);
+    // We are only testing a string field, a boolean field, a double field
+    assertEquals(
+        json.get("title").toString(),
+        "\"" + a.getTitle() + "\"",
```

Comment:
Il est beaucoup plus utile et simple de mettre le résultat attendu dans un fichier dans vos ressources et de comparer les deux.

Reply:
Fix in cda51eb1c7a8b88cc20ebbf50d26e8d681aacf75